confutils.nim

```diff
@@ -422,6 +508,95 @@ proc load*(Configuration: type,
     activeCmds.add cmd
     rejectNextArgument = not cmd.hasArguments
 
+  type
+    ArgKindFilter = enum
+      longForm
+      shortForm
+
+  proc showMatchingOptions(cmd: CommandPtr, prefix: string, filterKind: set[ArgKindFilter]) =
+    var matchingOptions: seq[OptionDesc]
+
+    if len(prefix) > 0:
+      # Filter the options according to the input prefix
+      for opt in cmd.options:
+        if longForm in filterKind:
+          if len(opt.name) > 0 and startsWithIgnoreStyle(opt.name, prefix):
+            matchingOptions.add(opt)
+        if shortForm in filterKind:
+          if len(opt.shortform) > 0 and
+            startsWithIgnoreStyle(opt.shortform, prefix):
+            matchingOptions.add(opt)
+    else:
+      matchingOptions = cmd.options
+
+    for opt in matchingOptions:
+      # The trailing '=' means the switch accepts an argument
+      let trailing = if opt.typename != "bool": "=" else: ""
+
+      if longForm in filterKind:
+        stdout.writeLine("--", opt.name, trailing)
+      if shortForm in filterKind:
+        stdout.writeLine('-', opt.shortform, trailing)
+
+  let completion = splitCompletionLine()
+  # If we're not asked to complete a command line the result is an empty list
+  if len(completion) != 0:
+    var cmdStack = @[rootCmd]
+    # Try to understand what the active chain of commands is without parsing the
+    # whole command line
+    for tok in completion[1..^1]:
+      if not tok.startsWith('-'):
+        let subCmd = findSubcommand(cmdStack[^1], tok)
+        if subCmd != nil: cmdStack.add(subCmd)
+
+    let cur_word = completion[^1]
```

Comment:
What about the cursor position btw? Is this reported by Bash somehow? I may be trying to complete something near the start of the line.

Reply:
`COMP_POINT` reports the cursor position, we cut the input line (`COMP_LINE`) at that point and then parse it.

Reply:
Ah, I've missed that detail, thanks.